tests/test_dummy_animal_module.py

```diff
@@ -0,0 +1,61 @@
+"""Test module for dummy_animal_module.py.
+
+This module tests the functionality of dummy_animal_module.py
+"""
+
+# Problems:
+# handling string returns for details
+# handling initialization of animals with zero energy
+# test that eat can't drive energy negative
+
+import virtual_rainforest.animals.dummy_animal_module as am
+
+
+class TestPlant:
+    """Test Plant class."""
+
+    def test_plant_growth0(self):
+        """Testing plant_growth at 100% energy."""
+        p = am.Plant("tree", 10.0, 1)
+        print("testing plant growth : plant")
+        p.plant_growth()
+        assert p.energy == 1000
+
+    def test_plant_growth1(self):
+        """Testing plant_growth at 100% energy."""
+        p = am.Plant("tree", 10.0, 1)
+        print("testing plant growth : plant")
+        p.plant_growth()
+        assert p.energy == 1000
+
+    def test_plant_growth2(self):
+        """Testing plant_growth at 50% energy."""
+        p = am.Plant("tree", 10.0, 1)
+        print("testing plant growth : plant")
+        p.energy = 500
+        p.plant_growth()
+        assert p.energy == 750
+
+    def test_plant_growth3(self):
+        """Testing plant_growth at 0% energy."""
+        p = am.Plant("tree", 10.0, 1)
+        print("testing plant growth : plant")
+        p.energy = 0
+        p.plant_growth()
+        assert p.energy == 0
```

Comment:
Instead of the separate `test_plant_growth*` functions, I'd use parameterised testing, as e.g. @jacobcook1995 has done here: https://github.com/ImperialCollegeLondon/virtual_rainforest/blob/develop/tests/test_main.py

That way you can test lots of different input/output combinations succinctly.

Reply:
The other think that is useful (but only marginally shorter here) is to use a `pytest.feature` for the repeated input:

```python
@pytest.feature
def plant_instance():
    from virtual_rainforest.animals.dummy_animal_module import Plant
    return am.Plant("tree", 10.0, 1)
```

And then tests can share that plant:

```python
def test_plant_growth3(plant_instance):
        """Testing plant_growth at 0% energy."""
        print("testing plant growth : plant")
        plant_instance.energy = 0
        plant_instance.plant_growth()
        assert plant_instance.energy == 0
```

Reply:
Also note that style of importing the `vr` functionality inside the test itself. That makes the ability to import the bit you want part of the test and an import fail only breaks the test that uses that functionality rather than the whole file.